newIDE/app/src/EventsSheet/ParameterFields/LeaderboardIdField.js

```diff
@@ -0,0 +1,147 @@
+// @flow
+import { Trans } from '@lingui/macro';
+import { type ParameterInlineRendererProps } from './ParameterInlineRenderer.flow';
+import React from 'react';
+import { type ParameterFieldProps } from './ParameterFieldCommons';
+import SelectField from '../../UI/SelectField';
+import SelectOption from '../../UI/SelectOption';
+import { TextFieldWithButtonLayout } from '../../UI/Layout';
+import RaisedButton from '../../UI/RaisedButton';
+import { type Leaderboard } from '../../Utils/GDevelopServices/Play';
+import LeaderboardContext from '../../Leaderboard/LeaderboardContext';
+import OpenInNew from '@material-ui/icons/OpenInNew';
+import { t } from '@lingui/macro';
+import Toggle from '../../UI/Toggle';
+import { LargeSpacer } from '../../UI/Grid';
+import LeaderboardDialog from '../../SceneEditor/LeaderboardDialog';
+import GenericExpressionField from './GenericExpressionField';
+
+const getLeaderboardNameByValue = (
+  leaderboards: ?Array<Leaderboard>,
+  value: string
+): string => {
+  if (!leaderboards) return value;
```

Comment:
I don't know if the whole uuid is necessary, maybe juste the first hyphen as it is done in the admin? I am not sure about that though

Reply:
This function will be used in inline display of parameter, I am not sure about changing the display of uuid specifically when there could be anything (like an expression instead)

Reply:
I think it's fine to return the whole value. We've not being able to identify a leaderboard id => we display everything.